ddtrace/contrib/internal/redis_utils.py

```diff
@@ -82,3 +68,36 @@ async def _run_redis_command_async(ctx: core.ExecutionContext, func, args, kwarg
         if redis_command not in ROW_RETURNING_COMMANDS:
             rowcount = None
         core.dispatch("redis.async_command.post", [ctx, rowcount])
+
+
+@contextmanager
+def _instrument_redis_execute_pipeline(pin, config_integration, cmds, instance):
+    cmd_string = resource = "\n".join(cmds)
+    if config_integration.resource_only_command:
+        resource = "\n".join([cmd.split(" ")[0] for cmd in cmds])
+
+    with core.context_with_data(
+        "redis.execute_pipeline",
+        span_name=schematize_cache_operation(redisx.CMD, cache_provider=redisx.APP),
+        resource=resource,
+        service=trace_utils.ext_service(pin, config_integration),
+        span_type=SpanTypes.REDIS,
+        pin=pin,
+    ) as ctx, ctx.span as span:
```

Comment:
can we add `redis.execute_pipeline` to the `_finish_span` context ended list, and then we don't need to reference `ctx.span` here at all. same below with `redis.command`